src/main/java/org/broadinstitute/hellbender/tools/walkers/conversion/GtfToBed.java

```diff
@@ -0,0 +1,288 @@
+ * This tool allows for the extraction of gene and transcript information from Gencode GTF files and
+ * outputs the data in BED format.</p>
+ *
+ *
+ * <p>The conversion process includes sorting entries
+ * by karyotype, providing flexibility in the selection of either gene or transcript level
+ * data, and an option to only use basic tags. It ensures that the BED output is sorted and formatted correctly for subsequent use.
+ * Note that it has been tested for both human and mouse Gencode GTFs. </p>
+ *
+ * <h3>Usage examples</h3>
+ * <p>Example commands to run GtfToBed for typical scenarios:</p>
+ *
+ * <h4>(i) Convert GTF to BED with gene level data</h4>
+ * <p>This mode extracts and converts gene data from the input GTF file to BED format:</p>
+ *
+ * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T False \
+ *     -B False \
+ *     -O output.bed \
+ * </pre>
+ *
+ * <h4>(ii) Convert GTF to BED with transcript level data</h4>
+ * <p>This mode extracts and converts transcript data from the input GTF file to BED format:</p>
+ *
+ * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T True \
+ *     -B False \
+ *     -O output.bed \
+ * </pre>
+ *
+ * <h4>(iii) Convert GTF to BED with transcript level data filtering for only those with the basic tag</h4>
+ *  * <p>This mode extracts and converts basic transcript data from the input GTF file to BED format:</p>
+ *  *
+ *  * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T True \
+ *     -B True \
+ *     -O output.bed \
+ *  * </pre>
+ */
+
+@CommandLineProgramProperties(
+        summary = "Converts Gencode GTF files to Bed file format with each row of bed file being either a gene or a transcript.",
+        oneLineSummary = "Gencode GTF to BED",
+        programGroup = ShortVariantDiscoveryProgramGroup.class
+)
+
+public class GtfToBed extends FeatureWalker<GencodeGtfFeature> {
+    public static final String SORT_BY_TRANSCRIPT_LONG_NAME = "SORT_BY_TRANSCRIPT";
+    public static final String SEQUENCE_DICTIONARY_LONG_NAME = "SEQUENCE_DICTIONARY";
+    public static final String SORT_BY_BASIC_LONG_NAME = "SORT_BY_BASIC";
+
+
+
+    @Argument(fullName = "GTF",
+            shortName = "G", doc = "Path to Gencode GTF file")
+    public GATKPath inputFile;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output BED file")
+    public GATKPath outputFile;
+
+    @Argument(fullName = SORT_BY_TRANSCRIPT_LONG_NAME, shortName = "T", doc = "Make each row of BED file sorted by transcript", optional = true)
+    public boolean sortByTranscript = false;
+
+    @Argument( fullName = SEQUENCE_DICTIONARY_LONG_NAME, shortName = "SD", doc = "Path to sequence dictionary")
```

Comment:
We generally don't point to the sequecne diciotnary and simply rely on the built-in GATK functionality to handle this. Since you are using a feature walker it already has a -R argument on the tool that already pulls in the sequence diciontary smartly from any source. 

You should be able to replace this argument and then use the method `getBestAvailibleSequenceDictionary()` to get thet sequence dictionary from the reference.

Reply:
This may cause problems for your mouse test, so you probably don't want the reference to be required. However most of the time folks in gatk use the -R input for the reference so that shoujld be a totally valid way to specify the input on the commandline. You can add a `CustomCommandLineValidation()` that enforces that there IS a sequence dictionary (from either -R or --sequence-dictionary) which is already specified in any GATKtool using this:
```
    @Argument(fullName = StandardArgumentDefinitions.SEQUENCE_DICTIONARY_NAME,
            shortName = StandardArgumentDefinitions.SEQUENCE_DICTIONARY_NAME,
            doc = "Use the given sequence dictionary as the master/canonical sequence dictionary.  Must be a .dict file.", optional = true, common = true)
    private GATKPath masterSequenceDictionaryFilename = null;
```

Reply:
Done